### PR TITLE
Tune up row size estimation in the data generator

### PR DESCRIPTION
**Pull request by vuule**
## Description

- Make string offsets a part of the strings column size;
- Fix erroneous "last element" inclusion in list columns;
- Minimize rounding errors by switching to double for the average row size;
- Account for null frequency for columns that don't store null elements (strings, lists);
- Account for the null masks size.

With these changes, actual table size should be much closer to the requested value. Tested indirectly through Parquet file size in benchmarks.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/common/generate_input.cu [hunk 1]
```diff
@@ -90,33 +90,61 @@ T get_distribution_mean(distribution_params<T> const& dist)
...
 template <>
-size_t non_fixed_width_size<cudf::string_view>(data_profile const& profile)
+double non_fixed_width_size<cudf::string_view>(data_profile const& profile)
 {
-  auto const dist = profile.get_distribution_params<cudf::string_view>().length_params;
-  return get_distribution_mean(dist);
+  auto const dist           = profile.get_distribution_params<cudf::string_view>().length_params;
+  auto const null_mask_size = profile.get_null_probability().has_value() ? 1. / 8 : 0.;
+  return get_distribution_mean(dist) * profile.get_valid_probability() + sizeof(cudf::size_type) +
+         null_mask_size;
```
Comment (davidwendt) [inline cpp/benchmarks/common/generate_input.cu:147]: This accounts for INT32 offsets but not INT64 ones. 
I'm not sure if that will be a necessary distinction here.

Comment (vuule) [reply cpp/benchmarks/common/generate_input.cu:147]: Yeah, I don't have a way to determine if INT64 will be used in the column. Number of rows/chars is not known at this time.

File: cpp/benchmarks/common/generate_input.cu [hunk 2]
```diff
@@ -90,33 +90,66 @@ T get_distribution_mean(distribution_params<T> const& dist)
...
+{
+  // Estimated average number of children in the hierarchy;
+  auto const num_children_avg = std::pow(num_leaf_columns, 1. / num_lvls);
+  // Minimum number of children columns for any column in the hierarchy
+  int const num_children_min = std::floor(num_children_avg);
+  // Maximum number of children columns for any column in the hierarchy
+  int const num_children_max = num_children_min + 1;
+
+  // Minimum number of columns needed so that their number of children does not exceed the maximum
+  int const min_for_current_nesting = std::ceil((double)num_leaf_columns / num_children_max);
```
Comment (bdice) [inline cpp/benchmarks/common/generate_input.cu:111]:
```suggestion
  int const min_for_current_nesting = std::ceil(static_cast<double>(num_leaf_columns) / num_children_max);
```

**Review by bdice — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Pull request merged**